src/link.c

```
@@ -1177,9 +1178,17 @@ long Link_download(Link *link, char *output_buf, size_t req_size, off_t offset)
         ts.data = NULL;
         ts.type = DATA;
         ts.transferring = 1;
+        ts.cache_ptr = cf;
+
+        if (cf) {
+            PTHREAD_MUTEX_LOCK(&cf->dl_lock);
+            cf->active_dl_ts = &ts;
```

Cppcheck (2.20.0)

[error] Address of local auto-variable assigned to a function parameter.

(autoVariables)